modules/andes-core/broker/src/main/java/org/wso2/andes/mqtt/MQTTMessageContext.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2005-2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.andes.mqtt;
+
+import io.netty.channel.Channel;
+import org.wso2.andes.kernel.distruptor.inbound.PubAckHandler;
+
+import java.nio.ByteBuffer;
+
+import static org.dna.mqtt.wso2.AndesMQTTBridge.QOSLevel;
+
+/**
+ * Builder pattern to contain parameter values defined through the protocol engine
+ */
+public class MQTTMessageContext {
```

Comment:
MQTTMessage extends AndesMessage which includes attributes generic to both protocols, MQTTMessageContext is created at the bridge where all the protocol specific attributes are being built which will be shared across the layers.